crates/rethnet_eth/src/remote/eth.rs

```diff
@@ -172,19 +172,37 @@ impl TryFrom<Transaction> for (SignedTransaction, Address) {
         };
 
         let transaction = match value.transaction_type {
-            0 => SignedTransaction::Legacy(LegacySignedTransaction {
-                nonce: value.nonce,
-                gas_price: value.gas_price,
-                gas_limit: value.gas.to(),
-                kind,
-                value: value.value,
-                input: value.input,
-                signature: Signature {
-                    r: value.r,
-                    s: value.s,
-                    v: value.v,
-                },
-            }),
+            0 => {
+                if value.v > 36 {
```

Comment:
I saw this check in two places, maybe we could add a function/method for deciding whether `v` is an EIP-155 value?

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4298/commits/99653bcb79aee565cae47c479caa4a7c03121da9